network-api/networkapi/wagtailpages/templatetags/blog_tags.py

```diff
@@ -0,0 +1,14 @@
+from django import template
+
+register = template.Library()
+
+
+@register.simple_tag(takes_context=True)
+def get_root_or_page(context):
+    root = context.get('root', None)
+    page = context.get('page', None)
+
+    if root:
+        return root.specific
+    else:
+        return page.get_parent().specific
```

Comment:
we technically don't know whether `page` is a real thing, so we probably want an `elif page: ... else: print("missing both root and page in get_root_or_page")`